### ISSUE TITLE
panic: not implemented from --mode validate

### ISSUE THREAD

**Issue by mtalbot** [envoyproxy/envoy#15902]
*Title*: Admin API TAP sockets fail on validate

*Description*:
When validating a static config file using `--mode validate` with tap transport sockets wrapping any other transport socket (tried with both raw buffer and tls) then envoy crashes with a `panic: not implemented` from https://github.com/envoyproxy/envoy/blob/main/source/server/config_validation/admin.cc#L13

*Repro steps*:
run envoy in validate mode with an admin api tap transport socket

```
[2021-04-09 14:35:15.911][35383][critical][assert] [source/server/config_validation/admin.cc:13] panic: not implemented
  [2021-04-09 14:35:15.911][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:104] Caught Aborted, suspect faulting address 0x8a37
  [2021-04-09 14:35:15.911][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
  [2021-04-09 14:35:15.911][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:92] Envoy version: d6a4496e712d7a2335b26e2f76210d5904002c26/1.17.1/Clean/RELEASE/BoringSSL
  [2021-04-09 14:35:15.912][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #0: __restore_rt [0x7f0ad4086980]
  [2021-04-09 14:35:15.918][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #1: Envoy::Extensions::Common::Tap::AdminHandler::AdminHandler() [0x56291892ab7d]
  [2021-04-09 14:35:15.925][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #2: std::__1::__function::__func<>::operator()() [0x56291892c850]
  [2021-04-09 14:35:15.932][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #3: Envoy::Singleton::ManagerImpl::get() [0x56291968bc2c]
  [2021-04-09 14:35:15.939][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #4: Envoy::Singleton::Manager::getTyped<>() [0x56291892a97e]
  [2021-04-09 14:35:15.946][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #5: Envoy::Extensions::Common::Tap::AdminHandler::getSingleton() [0x56291892a867]
  [2021-04-09 14:35:15.953][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #6: Envoy::Extensions::Common::Tap::ExtensionConfigBase::ExtensionConfigBase() [0x562918928df3]
  [2021-04-09 14:35:15.961][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #7: Envoy::Extensions::TransportSockets::Tap::TapSocketFactory::TapSocketFactory() [0x562918928688]
  [2021-04-09 14:35:15.967][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #8: Envoy::Extensions::TransportSockets::Tap::UpstreamTapSocketConfigFactory::createTransportSocketFactory() [0x5629189228cf]
  [2021-04-09 14:35:15.974][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #9: Envoy::Upstream::createTransportSocketFactory() [0x56291984865d]
  [2021-04-09 14:35:15.981][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #10: Envoy::Upstream::ClusterImplBase::ClusterImplBase() [0x562919848b9b]
  [2021-04-09 14:35:15.988][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #11: Envoy::Upstream::StrictDnsClusterImpl::StrictDnsClusterImpl() [0x56291986c86b]
  [2021-04-09 14:35:15.995][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #12: std::__1::__shared_ptr_emplace<>::__shared_ptr_emplace<>() [0x56291986f556]
  [2021-04-09 14:35:16.003][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #13: Envoy::Upstream::StrictDnsClusterFactory::createClusterImpl() [0x56291986da22]
  [2021-04-09 14:35:16.011][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #14: Envoy::Upstream::ClusterFactoryImplBase::create() [0x56291986b1ca]
  [2021-04-09 14:35:16.018][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #15: Envoy::Upstream::ClusterFactoryImplBase::create() [0x56291986a5ee]
  [2021-04-09 14:35:16.025][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #16: Envoy::Upstream::ProdClusterManagerFactory::clusterFromProto() [0x5629196b158d]
  [2021-04-09 14:35:16.032][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #17: Envoy::Upstream::ClusterManagerImpl::loadCluster() [0x5629196a7492]
  [2021-04-09 14:35:16.039][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #18: Envoy::Upstream::ClusterManagerImpl::ClusterManagerImpl() [0x5629196a4a55]
  [2021-04-09 14:35:16.047][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #19: Envoy::Upstream::ValidationClusterManagerFactory::clusterManagerFromProto() [0x56291963510d]
  [2021-04-09 14:35:16.054][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #20: Envoy::Server::Configuration::MainImpl::initialize() [0x56291992736f]
  [2021-04-09 14:35:16.061][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #21: Envoy::Server::ValidationInstance::initialize() [0x56291962f3ab]
  [2021-04-09 14:35:16.068][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #22: Envoy::Server::ValidationInstance::ValidationInstance() [0x56291962e25f]
  [2021-04-09 14:35:16.075][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #23: Envoy::Server::validateConfig() [0x56291962d695]
  [2021-04-09 14:35:16.084][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #24: Envoy::MainCommonBase::run() [0x562917e5be5d]
  [2021-04-09 14:35:16.091][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #25: Envoy::MainCommon::main() [0x562917e5c627]
  [2021-04-09 14:35:16.097][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #26: main [0x562917e5a9dc]
  [2021-04-09 14:35:16.098][35383][critical][backtrace] [bazel-out/k8-opt/bin/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #27: __libc_start_main [0x7f0ad3ca4bf7]
```

```yaml
  - connect_timeout: 5s
    lb_policy: round_robin
    load_assignment:
      cluster_name: service
      endpoints:
      - lb_endpoints:
          endpoint:
            address:
              socket_address:
                address: "service"
                port_value: 443
    name: api-gateway
    transport_socket:
      name: envoy.transport_sockets.tap
      typed_config:
        '@type': type.googleapis.com/envoy.extensions.transport_sockets.tap.v3.Tap
        common_config:
          admin_config:
            config_id: api-gateway
        transport_socket:
          name: envoy.transport_sockets.tls
          typed_config:
            '@type': type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.UpstreamTlsContext
            allow_renegotiation: true
            common_tls_context:
              tls_certificate_sds_secret_configs:
                name: client-certificate
                sds_config:
                  path: /etc/envoy/xds/sds-client-certificate.yaml
                  resource_api_version: V3
              tls_params:
                tls_minimum_protocol_version: TLSv1_2
              validation_context_sds_secret_config:
                name: client-validation
                sds_config:
                  path: /etc/envoy/xds/sds-client-validation.yaml
                  resource_api_version: V3
            sni: "service"
    type: STRICT_DNS
```

**Comment by snowp** [comment]
Thanks for the report, this seems like a clear bug so marking help wanted.

**Comment by davinci26** [comment]
For some reason I think my change here is the smoking gun, I will try to take a look today. Feel free to assign to me now as I do the triaging

**Comment by davinci26** [comment]
I got hang-up with some other things on Friday so I got around to the triaging today.

The regression is coming from https://github.com/envoyproxy/envoy/commit/e7a0f7c891bfcdb84c7b7d03bb5b48f1f148a47e

The root of the issue is that in order to generate the warning regarding using Admin tapping with `UDS` sockets we request the socket from the admin interface. This throws for the config_validation `admin` instance.

I will investigate a bit more to see what would be the best way to fix this and I will push a PR.

**Comment by slovdahl** [comment]
Any chance the fix in #15932 will be backported to the 1.18 release branch? FWIW, I think this is also broken in 1.17.

Seen on 1.17.1.p0.gd6a4496-1p74.gbb8060d:

```
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.079][10229][info][main] [external/envoy/source/server/server.cc:634] runtime: layers:
apr 16 08:37:51 lab envoy[10229]:   - name: static_layer_0
apr 16 08:37:51 lab envoy[10229]:     static_layer:
apr 16 08:37:51 lab envoy[10229]:       overload:
apr 16 08:37:51 lab envoy[10229]:         global_downstream_max_connections: 50000
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.079][10229][info][config] [external/envoy/source/server/configuration_impl.cc:125] loading tracing configuration
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.079][10229][info][config] [external/envoy/source/server/configuration_impl.cc:85] loading 0 static secret(s)
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.079][10229][info][config] [external/envoy/source/server/configuration_impl.cc:91] loading 2 cluster(s)
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.080][10229][info][config] [external/envoy/source/server/configuration_impl.cc:95] loading 2 listener(s)
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.085][10229][critical][assert] [external/envoy/source/server/config_validation/admin.cc:13] panic: not implemented
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.085][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:104] Caught Aborted, suspect faulting address 0x3e6000027f5
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.085][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:91] Backtrace (use tools/stack_decode.py to get line numbers):
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.085][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:92] Envoy version: d6a4496e712d7a2335b26e2f76210d5904002c26/1.17.1/clean-getenvoy-bb8060d-envoy/RELEASE/BoringSSL
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.085][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #0: __restore_rt [0x7f932f11e390]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.091][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #1: Envoy::Extensions::Common::Tap::AdminHandler::AdminHandler() [0x56527b5174d0]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.096][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #2: std::__1::__function::__func<>::operator()() [0x56527b519260]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.102][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #3: Envoy::Singleton::ManagerImpl::get() [0x56527c2d6a0b]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.107][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #4: Envoy::Singleton::Manager::getTyped<>() [0x56527b51729e]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.112][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #5: Envoy::Extensions::Common::Tap::AdminHandler::getSingleton() [0x56527b517187]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.117][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #6: Envoy::Extensions::Common::Tap::ExtensionConfigBase::ExtensionConfigBase() [0x56527b5156fc]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.122][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #7: Envoy::Extensions::HttpFilters::TapFilter::FilterConfigImpl::FilterConfigImpl() [0x56527ad6dade]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.128][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #8: Envoy::Extensions::HttpFilters::TapFilter::TapFilterFactory::createFilterFactoryFromProtoTyped() [0x56527ad6a795]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.133][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #9: Envoy::Extensions::HttpFilters::Common::FactoryBase<>::createFilterFactoryFromProto() [0x56527ad6ab5f]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.138][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #10: Envoy::Extensions::NetworkFilters::HttpConnectionManager::HttpConnectionManagerConfig::processFilter() [0x56527c5a2be1]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.143][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #11: Envoy::Extensions::NetworkFilters::HttpConnectionManager::HttpConnectionManagerConfig::HttpConnectionManagerConfig() [0x56527c5a0ca7]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.148][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #12: Envoy::Extensions::NetworkFilters::HttpConnectionManager::HttpConnectionManagerFilterConfigFactory::createFilterFactoryFromProtoTyped() [0x56527c59e852]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.153][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #13: Envoy::Extensions::NetworkFilters::Common::FactoryBase<>::createFilterFactoryFromProto() [0x56527c5a6116]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.158][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #14: Envoy::Server::ProdListenerComponentFactory::createNetworkFilterFactoryList_() [0x56527c558d78]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.164][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #15: Envoy::Server::ValidationInstance::createNetworkFilterFactoryList() [0x56527c279134]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.169][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #16: Envoy::Server::ListenerFilterChainFactoryBuilder::buildFilterChainInternal() [0x56527c5648dd]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.174][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #17: Envoy::Server::ListenerFilterChainFactoryBuilder::buildFilterChain() [0x56527c564729]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.179][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #18: Envoy::Server::FilterChainManagerImpl::addFilterChains() [0x56527c56e3cc]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.184][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #19: Envoy::Server::ListenerImpl::buildFilterChains() [0x56527c550b56]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.190][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #20: Envoy::Server::ListenerImpl::ListenerImpl() [0x56527c54e522]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.195][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #21: Envoy::Server::ListenerManagerImpl::addOrUpdateListenerInternal() [0x56527c55f691]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.200][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #22: Envoy::Server::ListenerManagerImpl::addOrUpdateListener() [0x56527c55e411]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.205][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #23: Envoy::Server::Configuration::MainImpl::initialize() [0x56527c5895f0]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.211][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #24: Envoy::Server::ValidationInstance::initialize() [0x56527c2776aa]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.216][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #25: Envoy::Server::ValidationInstance::ValidationInstance() [0x56527c27640d]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.221][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #26: Envoy::Server::validateConfig() [0x56527c2757bd]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.226][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #27: Envoy::MainCommonBase::run() [0x56527aa0b62d]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.231][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #28: Envoy::MainCommon::main() [0x56527aa0be27]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.237][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #29: main [0x56527aa0a13c]
apr 16 08:37:51 lab envoy[10229]: [2021-04-16 08:37:51.237][10229][critical][backtrace] [bazel-out/k8-opt/bin/external/envoy/source/server/_virtual_includes/backtrace_lib/server/backtrace.h:96] #30: __libc_start_main [0x7f932ed63830]
```
